components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -89,6 +103,63 @@ - (void)setSelectedItem:(UITabBarItem *)selectedItem {
   }
 
   _selectedItem = selectedItem;
+  if (itemIndex == NSNotFound) {
+    return;
+  }
+
+  UIView *itemView = self.itemViews[itemIndex];
+  if ([itemView isKindOfClass:[MDCTabBarViewItemView class]]) {
+    MDCTabBarViewItemView *selectedItemView = (MDCTabBarViewItemView *)itemView;
+    selectedItemView.iconImageView.image = selectedItem.image;
+  }
+}
+
+#pragma mark - Key-Value Observing (KVO)
+
+- (void)addObserversToTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item addObserver:self
+           forKeyPath:kImageKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+    [item addObserver:self
+           forKeyPath:kTitleKeyPath
+              options:NSKeyValueObservingOptionNew
+              context:kKVOContextMDCTabBarView];
+  }
+}
+
+- (void)removeObserversFromTabBarItems {
+  for (UITabBarItem *item in self.items) {
+    [item removeObserver:self forKeyPath:kImageKeyPath context:kKVOContextMDCTabBarView];
+    [item removeObserver:self forKeyPath:kTitleKeyPath context:kKVOContextMDCTabBarView];
+  }
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context {
+  if (context == kKVOContextMDCTabBarView && object) {
+    NSUInteger indexOfObject = [self.items indexOfObject:object];
+    if (indexOfObject == NSNotFound) {
+      return;
+    }
+    // Don't try to update custom views
+    UIView *updatedItemView = self.itemViews[indexOfObject];
+    if (![updatedItemView isKindOfClass:[MDCTabBarViewItemView class]]) {
+      return;
+    }
+    MDCTabBarViewItemView *tabBarItemView = (MDCTabBarViewItemView *)updatedItemView;
+
+    if ([keyPath isEqualToString:kImageKeyPath]) {
+      tabBarItemView.iconImageView.image = change[NSKeyValueChangeNewKey];
```

Comment:
Does it make sense to do more validation of the input before assigning it to the image? What if the value is not an image - will it crash?

Reply:
I think if someone manages to assign a non-UIImage to the UIImage property, it should crash.  That would be developer error and apple documents those as justifying a crash.